simpleflow/swf/process/worker/base.py

```diff
@@ -261,6 +261,46 @@ def spawn_kubernetes_job(poller, swf_response):
     job.schedule()
 
 
+def reap_process_tree(pid, wait_timeout=0.3):
```

Comment:
hey @ybastide I think this `wait_timeout` should actually be specific to `SIGTERM`.  A few questions: 

- Because it's a 'graceful exit' criteria, should it be configurable (via an env var or through simpleflow's settings subsystem)? I think 'definitely' since different applications may want different timeouts if they implement termination handlers.
- Thoughts on the default timeout for SIGTERM? It should ideally be longer than 0.3 sec (like 3-5 seconds), but I'm guessing that this value will cause existing applications that don't handle it to pause for longer than expected before getting killed.

Reply:
I agree on all your counts :slightly_smiling_face:
* no need to use `wait_timeout` after the kills
* best to define as a setting
* a default of 3 seconds is fine by me

Reply:
Done. Following up with test coverage. 

Let me know how you feel about the implementation around settings. It's a little hard to tell what's currently in use, and also what counts as a 'constant' vs a setting.

(Also, sorry for the long delay on follow-up).